pkg/operator/ceph/cluster/identity/identity.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package identity
+
+import (
+	"fmt"
+	"sync"
+)
+
+// An Identity contains all of the information that Ceph daemons need to share regarding the
+// identity of the cluster to which they belong.
+type Identity struct {
+	fsid string
+	m    sync.Mutex
+}
+
+// BlankIdentity returns a blank Identity. A blank Identity's get methods will return an error until
+// the Identity is Establish-ed.
+func BlankIdentity() *Identity { return &Identity{} }
+
+// helper method that doesn't try to lock/unlock the mutex
+func (i *Identity) isEstablished() bool {
+	return i.fsid != ""
+}
+
+// IsEstablished returns true if the identity has been established. False otherwise.
+func (i *Identity) IsEstablished() bool {
+	i.m.Lock()
+	defer i.m.Unlock()
```

Comment:
Mostly I was thinking of the various controller processes that might be requesting the identity at the same time. It's true that the identity should already be established by the time multiple processes exist, though ... 

I could see about using a RWMutex https://golang.org/pkg/sync/#RWMutex so that it would only block if something were trying to write. That *shouldn't* happen, but I don't think it would hurt to assume someone might do something funny in the future.